### PR TITLE
Add cleanup workflow

### LINKED ISSUE #275: CI Migration: Delete test namespaces after PR is closed
## Description
Blocked by #270 
With each pull request creating a new namespace for each PR's deployment, this could quickly create several namespaces.  To avoid manually deleting namespaces, it would be great if namespaces were deleted after a PR is closed.  

## Notes
This could be implemented several ways: k8s job, Cloud Build preliminary step, Github webhook etc

### PR DESCRIPTION
**Pull request by WilliamG240**
Adds deletion of test namespace on cluster after closing of pull request.  Fixes #275

### CODE REVIEW

File: .github/workflows/cleanup.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Clean up deployment"
+on:
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+    types: closed
```
Comment (murog) [inline .github/workflows/cleanup.yaml:21]: didn't know this was an option!!🏆

Comment (WilliamG240) [reply .github/workflows/cleanup.yaml:21]: So this deleted the namespace when the PR was closed (i closed it multiple times to test on this PR)

Comment (murog) [reply .github/workflows/cleanup.yaml:21]: ah, thanks for clearing that up!

**Comment by murog** [comment]
LGTM 🚀  thanks for looking into actions running on closed pull requests! this is a lot more maintainable that a github bot (-:

**Pull request merged**